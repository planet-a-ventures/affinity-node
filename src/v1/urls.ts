/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-a-specific-list) for more info.
 */
export const listsUrl = (list_id?: number) =>
    list_id ? `/lists/${encodeURIComponent(list_id)}` : '/lists'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-a-specific-list-entry) for more info.
 */
export const listEntriesUrl = (list_id: number, list_entry_id?: number) =>
    listsUrl(list_id) + (
        list_entry_id
            ? `/list-entries/${encodeURIComponent(list_entry_id)}`
            : '/list-entries'
    )

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#rate-limit) for more info.
 */
export const rateLimitUrl = () => '/rate-limit'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#whoami) for more info.
 */
export const whoAmIUrl = () => '/auth/whoami'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#fields) for more info.
 */
export const fieldsUrl = (field_id?: number) =>
    field_id ? `/fields/${encodeURIComponent(field_id)}` : '/fields'
