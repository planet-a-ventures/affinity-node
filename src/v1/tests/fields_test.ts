import { assertSnapshot } from '@std/testing/snapshot.ts'
import { afterEach, beforeEach, describe, it } from '@std/testing/bdd.ts'

import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { Affinity } from '../index.ts'
import { getRawFixture } from './get_raw_fixture.ts'
import { apiKey, isLiveRun } from './env.ts'
import { fieldsUrl } from '../urls.ts'

describe('fields', () => {
    let mock: MockAdapter
    let affinity: Affinity

    beforeEach(() => {
        if (!isLiveRun()) {
            mock = new MockAdapter(axios, { onNoMatch: 'throwException' })
        }
        affinity = new Affinity(apiKey() || 'api_key')
    })
    afterEach(() => {
        mock?.reset()
    })

    describe('all', () => {
        it('fetches all fields', async (t) => {
            const fixture = await getRawFixture('fields/all.raw.response.json')

            mock.onGet(fieldsUrl()).reply(200, fixture)

            const res = await affinity.fields.all()
            await assertSnapshot(t, res)
        })
    })
})
