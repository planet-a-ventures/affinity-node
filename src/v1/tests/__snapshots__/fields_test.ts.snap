export const snapshot = {};

snapshot[`fields > all > fetches all fields 1`] = `
[
  {
    allows_multiple: false,
    dropdown_options: [],
    enrichment_source: "none",
    id: 1234,
    list_id: 11,
    name: "[Deals] Amount",
    track_changes: true,
    value_type: 3,
  },
  {
    allows_multiple: false,
    dropdown_options: [],
    enrichment_source: "none",
    id: 5678,
    list_id: 16,
    name: "[Events] Amount",
    track_changes: true,
    value_type: 3,
  },
  {
    allows_multiple: false,
    dropdown_options: [],
    enrichment_source: "dealroom",
    id: 4321,
    list_id: 18,
    name: "[Companies] Description",
    track_changes: false,
    value_type: 6,
  },
]
`;
